src/app/pages/review-attributes/review-attributes.page.ts

```diff
@@ -63,65 +56,64 @@ export class ReviewAttributesPage implements OnInit {
 
   onApprovePhraseBtnClick() {
     const self = this;
-
+    const phrase = self._reviewAttributesModelService.get();
+    var reviewId = phrase.id; // get from modelservice
+    var reasonID = 1;
+    self._reviewAttributesModelService.saveReviewAttributes(reviewId,reasonID);
     self._alertService.show({
-      header: "Message approved!",
-      buttons: [
-        {
-          text: "OK",
-          role: "cancel",
-          handler: () => {},
-        },
-      ],
-    });
+    header: 'Message approved!',
+    buttons: [{
+      text: 'OK', role: 'cancel',
+      handler: () => {
+      }
+      }]
+    })
 
     this.phraseToBeReviewed = "";
     this.getNextPhraseButtonDisabled = false;
     this.approveAndRejectButtonDisplayed = false;
   }
 
   onGetNextPhraseBtnClick() {
+
     this.getNextPhraseToBeReviewed();
   }
 
   getNextPhraseToBeReviewed() {
-    const self = this;
-    const phraseTBR =
-      this._reviewAttributesModelService.getNextReviewEligiblePhrase();
-    if (phraseTBR) {
-      const phraseTBRAsString = this.getAttrString(phraseTBR);
+    this._reviewAttributesModelService.getNewPhrase();
+    const wait = ms => new Promise(resolve => setTimeout(resolve, ms));
+    console.log("newphrase loading ...");
+    wait(1000).then(() => {
```

Comment:
It doesn't feel like this wait() is necessary. What's your rationale here?

Reply:
there is a delay from acquiring the next phrase from the backend call so the wait() is need to fill the variable with phrase before continue with code. otherwise it return undefined to review page when nextphrase button is click. This works but I don't know if there better way to implement the wait.